rct229/rulesets/ashrae9012019/section4/section4rule14.py

```diff
@@ -0,0 +1,103 @@
+            ruleset_section_title="Schedule - Setpoints",
+            standard_section="Section 3 Definitions",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.buildings[*].building_segments[*].zones[*].spaces[*]",
+        )
+
+    def create_data(self, context, data):
+        rmd_p = context.proposed
+
+        sch_ids_p = find_all(
+            f'$.buildings[*].building_segments[*].zones[*].spaces[*].miscellaneous_equipment[*][?(@.energy_type="ELECTRICITY")].multiplier_schedule',
+            rmd_p,
+        )
+        schedule_p = []
+        for scd_id_p in sch_ids_p:
+            schedule_p += find_all(f'$.schedules[*][?(@id = "{scd_id_p}")]', rmd_p)
+
+        return {"schedule_p": schedule_p}
+
+    class SpaceRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section4Rule14.SpaceRule, self).__init__(
+                rmrs_used=produce_ruleset_model_instance(
+                    USER=False, BASELINE_0=False, PROPOSED=True
+                ),
+                required_fields={
+                    "$": [
+                        "lighting_space_type",
+                        "miscellaneous_equipment",
+                        "floor_area",
+                    ],
+                },
+                fail_msg="The space has been classed as a computer room in terms of the lighting space type but the electronic data equipment power density does not appear to exceed 20 w/sf.",
+            )
+
+        def get_calc_vals(self, context, data=None):
+            space_p = context.proposed
+
+            schedule_p = data["schedule_p"]
+            floor_area_p = space_p["floor_area"]
+
+            total_space_misc_Wattage_including_multiplier_p = ZERO.POWER
+            if space_p["lighting_space_type"] == LIGHTING_SPACE.COMPUTER_ROOM:
+                for misc_equip_p in space_p["miscellaneous_equipment"]:
+                    if misc_equip_p["energy_type"] == ENERGY_SOURCE.ELECTRICITY:
+                        misc_equip_power_p = getattr_(
+                            misc_equip_p, "miscellaneous_equipment", "power"
+                        )
+                        mis_multiplier_id_p = getattr_(
+                            misc_equip_p,
+                            "miscellaneous_equipment",
+                            "multiplier_schedule",
+                        )
+
+                        total_space_misc_Wattage_including_multiplier_p += min(
+                            1,
+                            max(
+                                misc_equip_power_p
+                                * find(
+                                    schedule_p,
+                                    lambda d: d.get("id") == mis_multiplier_id_p,
+                                )
+                            ),
+                        )
```

Comment:
In RDS, it says `MAX(1, MAX(misc_p.multiplier_schedule.hourly_values)` and then `misc_p_total_Wattage = misc_p_power * misc_p_multiplier_value`.
The expression used about seems wrong.
This shall be revised to:
```python
misc_equip_power_p * max(1.0, max(getattr_(find(schedule_p, lambda d: d["id"] == mis_multiplier_id_p), "schedule", "hourly_values")))
```

You can also break it down into multiple expressions if this is getting too long to understand.